gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -1786,6 +1839,14 @@ gst_tensor_filter_stop (GstBaseTransform * trans)
   GstTensorFilterPrivate *priv;
   self = GST_TENSOR_FILTER_CAST (trans);
   priv = &self->priv;
+
+  if (priv->prop.suspend != 0) {
+    GST_OBJECT_LOCK (self);
+    nnstreamer_watchdog_destroy (priv->watchdog_h);
+    priv->watchdog_h = NULL;
+    GST_OBJECT_UNLOCK (self);
+  }
```

Comment:
You appear to create and destroy threads for every inference call. Doesn't incur too much overhead?

Can you keep the watchdos persistent and alter its behavior with signal? (imagine when you have a network with 10ms latency and calling it 100 times/sec.)

Reply:
Note that creating and destroying threads may take more than 10ms in embedded devices.

Reply:
> You appear to create and destroy threads for every inference call. Doesn't incur too much overhead?

The suspended mode is activated only in the playing state and does not change the state of the pipeline.
When the watchdog is triggered (timeout), only the framework is unloaded, thread is alive and the state remains playing.
So, watchdog thread is created when the pipeline goes to playing state, It is destroyed when the pipeline stops or destroyed.